src/sgame/botlib/bot_nav.cpp

```diff
@@ -408,25 +422,47 @@ void G_BotAddObstacle( const glm::vec3 &mins, const glm::vec3 &maxs, qhandle_t *
 
 		tempBox.maxs[ 0 ] += offset;
 		tempBox.maxs[ 2 ] += offset;
-		
+
 		// offset mins down by agent height so obstacles placed on ledges are handled correctly
 		tempBox.mins[ 1 ] -= params->walkableHeight;
 
 		nav->cache->addBoxObstacle( tempBox.mins, tempBox.maxs, &ref );
-		*obstacleHandle = ref;
+		obstacleHandles[i] = ref;
 	}
 }
 
-void G_BotRemoveObstacle( qhandle_t obstacleHandle )
+// We do lazy load navmesh when bots are added. The downside is that this means
+// map entities are loaded before the navmesh are. This workaround does keep
+// those obstacle (such as doors and buildables) in mind until when navmesh is
+// finally loaded, or generated.
+void BotAddSavedObstacles()
 {
-	for ( int i = 0; i < numNavData; i++ )
+	for ( std::pair<int, bbox_t> obstacle : savedObstacles )
```

Comment:
nit const std::pair<int, bbox_t>& or const auto&

Reply:
IIRC `std::pair<int, bbox_t>&` doesn't compile, somehow. This function is called once per game anyway so it doesn't really matter